ion/integration_test.go

```diff
@@ -0,0 +1,923 @@
+	"timestamp_0001-00-01.ion",
+	"timestamp_0001-00-01T.ion",
+	"timestamp_0001-00T.ion",
+	"timestamp_0001-01-00.ion",
+	"timestamp_0001-01-00T.ion",
+	"timestamp_10.ion",
+	"timestamp_11.ion",
+	"timestamp_5.ion",
+	"timestamp_6.ion",
+	"timestamp_7.ion",
+	"timestamp_8.ion",
+	"timestamp_9.ion",
+	"type_3_length_0.10n",
+	"year_3.ion",
+}
+
+var equivsSkipList = []string{
+	"annotatedIvms.ion",
+	"clobs.ion",
+	"localSymbolTableAppend.ion",
+	"localSymbolTableNullSlots.ion",
+	"localSymbolTableWithAnnotations.ion",
+	"localSymbolTables.ion",
+	"localSymbolTablesValuesWithAnnotations.ion",
+	"nonIVMNoOps.ion",
+	"sexps.ion",
+	"stringUtf8.ion",
+	"strings.ion",
+	"structsFieldsDiffOrder.ion",
+	"structsFieldsRepeatedNames.ion",
+	"systemSymbols.ion",
+	"systemSymbolsAsAnnotations.ion",
+	"timestampSuperfluousOffset.10n",
+	"timestamps.ion",
+	"timestampsLargeFractionalPrecision.ion",
+}
+
+var nonEquivsSkipList = []string{
+	"decimals.ion",
+	"documents.ion",
+	"floats.ion",
+	"floatsVsDecimals.ion",
+	"localSymbolTableWithAnnotations.ion",
+	"structs.ion",
+	"symbolTables.ion",
+	"symbolTablesUnknownText.ion",
+	"symbols.ion",
+	"timestamps.ion",
+}
+
+func TestBinaryRoundTrip(t *testing.T) {
+	readFilesAndTest(t, goodPath, binaryRoundTripSkipList, func(t *testing.T, path string) {
+		testBinaryRoundTrip(t, path)
+	})
+}
+
+func TestTextRoundTrip(t *testing.T) {
+	readFilesAndTest(t, goodPath, textRoundTripSkipList, func(t *testing.T, path string) {
+		testTextRoundTrip(t, path)
+	})
+}
+
+func TestLoadBad(t *testing.T) {
+	readFilesAndTest(t, badPath, malformedIonsSkipList, func(t *testing.T, path string) {
+		testLoadBad(t, path)
+	})
+}
+
+func TestEquivalency(t *testing.T) {
+	readFilesAndTest(t, equivsPath, equivsSkipList, func(t *testing.T, path string) {
+		testEquivalency(t, path, true)
+	})
+}
+
+func TestNonEquivalency(t *testing.T) {
+	readFilesAndTest(t, nonEquivsPath, nonEquivsSkipList, func(t *testing.T, path string) {
+		testEquivalency(t, path, false)
+	})
+}
+
+// Execute round trip testing for BinaryWriter. Create a BinaryWriter, using the BinaryWriter create
+// a TextWriter and back to BinaryWriter again. Validate that the first and last Writers are equal.
+func testBinaryRoundTrip(t *testing.T, fp string) {
+	fileBytes := loadFile(t, fp)
+
+	// Make a binary writer from the file
+	buf := encodeAsBinaryIon(t, fileBytes)
+
+	// Re-encode binWriter's stream as text into a string builder
+	str := encodeAsTextIon(t, buf.String())
+
+	reader1 := NewReader(bytes.NewReader(buf.Bytes()))
+	reader2 := NewReader(strings.NewReader(str.String()))
+
+	for reader1.Next() {
+		i1 := readCurrentValue(t, reader1)
+		reader2.Next()
+		i2 := readCurrentValue(t, reader2)
+
+		if !i1.equal(i2) {
+			t.Errorf("Failed on %s round trip. Binary reader has %v "+
+				"where the value in Text reader is %v", fp, i1.value, i2.value)
+		}
+	}
+}
+
+// Execute round trip testing for TextWriter. Create a TextWriter, using the TextWriter creat a
```

Comment:
* It looks like these comments are out of date now. ("Validate that the first and last Writers are equal.")
* "creat" -> "create"